src/PythonMigrationViewExtension/PythonMigrationViewExtension.cs

```diff
@@ -295,66 +462,131 @@ private void RecomputeCPython3NotificationForWorkspace()
                 return;
             }
```

Comment:
For the code above, relying on `RecomputeEngineFlags` being called before this line to update `hasCPython3Engine` (which is defined elsewhere) feels fragile -- it’s possible that `RecomputeEngineFlags` could be skipped due to future changes. Since `RecomputeEngineFlags` is lightweight, let’s introduce a `HasEngine()` method and use that here instead:

```csharp
if (preferenceSettings == null)
{
  if (PythonEngineManager.Instance.HasEngine(PythonEngineManager.CPython3EngineName))
  {
    CurrentWorkspace.ShowCPythonNotifications = false;
    return;
  }
}
```